src/test/java/common/bankarskiSistem/unit/service/ConversionServiceUnitTests.java

```diff
@@ -0,0 +1,284 @@
+package common.bankarskiSistem.unit.service;
+
+import common.bankarskiSistem.model.Bank;
+import common.bankarskiSistem.model.Conversion;
+import common.bankarskiSistem.model.Currency;
+import common.bankarskiSistem.repository.ConversionRepository;
+import common.bankarskiSistem.service.ConversionService;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.MethodSource;
+import org.junit.jupiter.params.provider.NullSource;
+import org.junit.jupiter.params.provider.ValueSource;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.junit.jupiter.MockitoExtension;
+
+import java.util.Optional;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.junit.jupiter.api.Assertions.*;
+import static org.mockito.Mockito.*;
+
+@ExtendWith(MockitoExtension.class)
+public class ConversionServiceUnitTests {
+    @InjectMocks
+    private ConversionService conversionService;
+    @Mock
+    private ConversionRepository conversionRepository;
+
+
+    @ParameterizedTest
+    //
+    @MethodSource("common.bankarskiSistem.parametrized.ConversionServiceParameters#generateConversion")
+    public void whenValidId_thenConversionShouldBeFound(Conversion conversion) {
+        //Given
+        when(conversionRepository.findByIdConversion(conversion.getIdConversion()))
+                .thenReturn(Optional.of(conversion));
+
+        //When
+        Conversion found = conversionService.findByIdConversion(conversion.getIdConversion());
+
+        //Then
+        assertThat(found.getIdConversion()).isEqualTo(conversion.getIdConversion());
+    }
+
+    @ParameterizedTest
+    @ValueSource(ints = {1, 5})
+    public void whenInvalidId_thenThrowNullPointerException(Integer nonValidIdConversion) {
+        //Given
+        when(conversionRepository.findByIdConversion(nonValidIdConversion)).thenReturn(Optional.empty());
+
+        //When
+        Exception exception = assertThrows(NullPointerException.class,
```

Comment:
Mozes da zamijenis ovim, s tim sto nije potrebno ispisivati poruku ako ima samo jedan NullPointerException.

assertThatThrownBy(() -> conversionkService.findByIdConversion(nonValidIdConversion))
                .isInstanceOf(NullPointerException.class)
                .hasMessageContaining("The conversion dose not exist.");